### ISSUE TITLE
undefined reference to symbol 'log@@GLIBC_2.2.5'

### ISSUE THREAD

**Issue by jiandongjia** [twitter/heron#915]
hi guys:

I use the CentOS system.
When I build the IDEA project, the error  as 
```
INFO: From Linking heron/common/tests/cpp/basics/fileutils_unittest:
/usr/bin/ld: bazel-out/local_linux-fastbuild/genfiles/3rdparty/gperftools/lib/libtcmalloc.a(libtcmalloc_internal_la-sampler.o): undefined reference to symbol 'log@@GLIBC_2.2.5'
/usr/bin/ld: note: 'log@@GLIBC_2.2.5' is defined in DSO /lib64/libm.so.6 so try adding it to the linker command line
/lib64/libm.so.6: could not read symbols: Invalid operation
collect2: error: ld returned 1 exit status
ERROR: /root/heron/heron/common/tests/cpp/basics/BUILD:3:1: Linking of rule '//heron/common/tests/cpp/basics:fileutils_unittest' failed: gcc failed: error executing command /usr/bin/gcc -o bazel-out/local_linux-fastbuild/bin/heron/common/tests/cpp/basics/fileutils_unittest -B/usr/bin/ -Wl,-z,relro,-z,now -no-canonical-prefixes -pass-exit-codes '-Wl,--build-id=md5' ... (remaining 3 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
INFO: Elapsed time: 50.939s, Critical Path: 41.63s.
WARNING!!! - bazel build failed - intellij setup may not be consistent
Path is /root/heron
Generating IDEA project...
scripts/get_all_heron_paths.sh: line 70: JAVA_PATHS: unbound variable
Done. IDEA module file: heron.iml

Could not locate IntelliJ IDEA. Manually open /root/heron as a project in IDEA.
```
I try to modify the config/configure.ac, add the following variables, but no effect.
```
#export LDFLAGS="$LDFLAGS -lm"
#LD="$LD -lm"
```

**Comment by jiandongjia** [comment]
I modified get_all_heron_paths.sh file, add a parameter --config = centos, now works correctly.

![default](https://cloud.githubusercontent.com/assets/19700922/16071301/2feb9f68-330d-11e6-83e2-da2cd7e2d49d.jpg)

**Comment by kramasamy** [comment]
When we build manually we provide the config=<platform> but in IDEA you need to query the platform and accordingly pass the platform. 

for ubuntu, it is --config=ubuntu
for mac, it is --config=darwin
for centos, it is --config=centos

It will be great if you could provide a patch to fix this.

**Comment by billonahill** [comment]
That's a bug that we don't pass `--config=[os]` to bazel on that line. I suggest we fix by defaulting to `--config=darwin` (the most common case), but allowing that to be overridden with a flag.

**Comment by caofangkun** [comment]
The fix has been merged as #985 , please close this issue.

**Comment by maosongfu** [comment]
Fixed: #985
